### PR TITLE
Started on a statistics page

### PR DESCRIPTION
**Pull request by lajtmaN**
image

### CODE REVIEW

File: backend/server/router.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ func NewRouter() *gin.Engine {
 		{
 			stats := new(controllers.StatsController)
 			s.GET("/leaderboard", stats.GetLeaderboard)
-			s.GET("/player-history/:userId", stats.GetPlayerHistory)
+			s.GET("/player-history", stats.GetPlayerHistory)
```
Comment (bjarkehs) [inline backend/server/router.go:32]: Husk at opdatere kommentaren i stats_controller.go også

**Review by bjarkehs — CHANGES_REQUESTED** [review]
Remember to change the swagger comment `stats_controller.go` when changing the route in `router.go`

File: backend/controllers/stats_controller.go [hunk 2]
```diff
@@ -51,16 +51,28 @@ func (sc StatsController) GetLeaderboard(c *gin.Context) {
 //	@Accept			json
 //	@Produce		json
 //	@Success		200	{array}	view.PlayerHistoryDetails
-//	@Router			/stats/player-history/{userId} [get]
+//	@Router			/stats/player-history [get]
```
Comment (bjarkehs) [inline backend/controllers/stats_controller.go:54]: You need to also update the parameter from `path` to `query`

File: backend/controllers/stats_controller.go [hunk 3]
```diff
@@ -51,16 +51,28 @@ func (sc StatsController) GetLeaderboard(c *gin.Context) {
...
+	var userId *uint = nil
+	userIdParam := c.Param("userId")
+
+	if userIdParam != "" {
+		userIdParsed, err := strconv.ParseUint(userIdParam, 10, 64)
+		if err != nil {
+			c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{"error": "Invalid user ID"})
+			return
+		}
+		userIdUint := uint(userIdParsed) // Convert userIdParsed to uint
```
Comment (bjarkehs) [inline backend/controllers/stats_controller.go:70]: Is it not possible to merge these to?
```go
userId = &uint(userIdParsed)
```

**Review by bjarkehs — CHANGES_REQUESTED** [review]

File: backend/controllers/stats_controller.go (hunk 3 above)
Comment (bjarkehs) [reply backend/controllers/stats_controller.go:70]: Seemingly it is not 😅

**Review by bjarkehs — APPROVED** [review]
LGTM

**Review by madsthom — APPROVED** [review]
Kanon!

**Pull request merged**